python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,247 @@
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table {database_name}.{table_name} fail to be created") from e
+
+        glue_table = load_table_response[PROP_GLUE_TABLE]
+        return _convert_glue_to_iceberg(glue_table, io)
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        """Loads the table's metadata and returns the table instance.
+        You can also use this method to check for table existence using 'try catalog.table() except TableNotFoundError'
+        Note: This method doesn't scan data stored in the table.
+        Args:
+            identifier: Table identifier.
+        Returns:
+            Table: the table instance with its metadata
+        Raises:
+            NoSuchTableError: If a table with the name does not exist, or the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            load_table_response = self.glue.get_table(DatabaseName=database_name, Name=table_name)
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table does not exists: {table_name}") from e
+        loaded_table = load_table_response[PROP_GLUE_TABLE]
+        io = load_file_io(self.properties, loaded_table[PROP_GLUE_TABLE_PARAMETERS][PROP_METADATA_LOCATION])
+        return _convert_glue_to_iceberg(loaded_table, io)
+
+    def drop_table(self, identifier: Union[str, Identifier]) -> None:
+        raise NotImplementedError("currently unsupported")
+
+    def purge_table(self, identifier: Union[str, Identifier]) -> None:
+        self.drop_table(identifier)
+
+    def rename_table(self, from_identifier: Union[str, Identifier], to_identifier: Union[str, Identifier]) -> Table:
+        raise NotImplementedError("currently unsupported")
+
+    def create_namespace(self, namespace: Union[str, Identifier], properties: Properties = EMPTY_DICT) -> None:
+        raise NotImplementedError("currently unsupported")
+
+    def drop_namespace(self, namespace: Union[str, Identifier]) -> None:
+        raise NotImplementedError("currently unsupported")
+
+    def list_tables(self, namespace: Union[str, Identifier]) -> List[Identifier]:
+        raise NotImplementedError("currently unsupported")
+
+    def list_namespaces(self, namespace: Union[str, Identifier] = ()) -> List[Identifier]:
+        """List namespaces from the given namespace. If not given, list top-level namespaces from the catalog.
+        Returns:
+            List[Identifier]: a List of namespace identifiers
+        """
+        # Glue does not support hierarchical namespace, therefore return an empty list
+        if namespace:
+            return []
+        databases_response = self.glue.get_databases()
+        return [
+            self.identifier_to_tuple(database[PROP_GLUE_DATABASE_NAME])
```

Comment:
Why does this need to parse a name coming back from Glue?

Reply:
Thank you for your review. The names coming back from Glue are `string`. According to the interface definition: https://github.com/apache/iceberg/blob/336d18acdb7dc0e783861151e2e871524cef9f6d/python/pyiceberg/catalog/__init__.py#L258, we need to return a list of `Identifier`, which is a tuple defined as :https://github.com/apache/iceberg/blob/336d18acdb7dc0e783861151e2e871524cef9f6d/python/pyiceberg/typedef.py#L36
Hence, I use `identifier_to_tuple` to convert the names to `Identifer` so that they can meet the interface requirement. If I misunderstand some points, please let me know.